tests/Oryx.Integration.Tests/Node/NodePreRunCommandOrScriptTest.cs

```diff
@@ -0,0 +1,140 @@
+                .SetEnvironmentVariable(
+                    SdkStorageConstants.SdkStorageBaseUrlKeyName,
+                    SdkStorageConstants.DevSdkStorageBaseUrl)
+                .SetEnvironmentVariable(FilePaths.PreRunCommandEnvVarName, "\"touch test_pre_run.txt\"")
+                .AddCommand($"oryx create-script -appPath {appOutputDir} -bindPort {ContainerPort}")
+                .AddFileExistsCheck($"test_pre_run.txt")
+                .AddCommand($"rm test_pre_run.txt")
+                .AddCommand(DefaultStartupFilePath)
+                .ToString();
+
+            await EndToEndTestHelper.BuildRunAndAssertAppAsync(
+                appName,
+                _output,
+                new[] { volume },
+                _imageHelper.GetGitHubActionsBuildImage(),
+                "/bin/sh",
+                new[]
+                {
+                    "-c",
+                    buildScript
+                },
+                _imageHelper.GetRuntimeImage("node", "dynamic"),
+                ContainerPort,
+                "/bin/sh",
+                new[]
+                {
+                    "-c",
+                    runScript
+                },
+                async (hostPort) =>
+                {
+                    var data = await _httpClient.GetStringAsync($"http://localhost:{hostPort}/");
+                    Assert.Contains("Say It Again", data);
+                });
+        }
+
+        [Fact]
+        public async Task CanBuildAndRunApp_UsingPreRunScript_WithDynamicInstall()
+        {
+            // Arrange
+            var nodeVersion = NodeVersions.Node12Version;
+            var appName = "webfrontend";
+            var volume = CreateAppVolume(appName);
+            var appDir = volume.ContainerDir;
+            var appOutputDir = $"{appDir}/myoutputdir";
+            var buildScript = new ShellScriptBuilder()
+                .SetEnvironmentVariable(SettingsKeys.EnableDynamicInstall, true.ToString())
+                .SetEnvironmentVariable(
+                    SdkStorageConstants.SdkStorageBaseUrlKeyName,
+                    SdkStorageConstants.DevSdkStorageBaseUrl)
+                .AddCommand(
+                $"oryx build {appDir} --platform nodejs --language-version {nodeVersion} -o {appOutputDir}")
+                .ToString();
+            var runScript = new ShellScriptBuilder()
+                .SetEnvironmentVariable(
+                    SdkStorageConstants.SdkStorageBaseUrlKeyName,
+                    SdkStorageConstants.DevSdkStorageBaseUrl)
+                .SetEnvironmentVariable(SettingsKeys.EnableDynamicInstall, true.ToString())
+                .SetEnvironmentVariable(FilePaths.PreRunCommandEnvVarName, "./prerunscript.sh")
+                .AddCommand($"touch {appOutputDir}/prerunscript.sh")
+                .AddFileExistsCheck($"{appOutputDir}/prerunscript.sh")
+                .AddCommand($"echo \"touch test_pre_run.txt\" > {appOutputDir}/prerunscript.sh")
```

Comment:
can we test a script which is bit more complicated or rather which matches more with a regular bash/shell script? this is more like running a command rather than script to me.